### PR TITLE
Buddy Emotes and Emote cleanup

### PR DESCRIPTION
**Pull request by Zintixx**
Implemented Buddy Emotes and touched up the current emote handler and packet. Renamed EmotionHandler to EmoteHandler.

### CODE REVIEW

File: MapleServer2/Types/BuddyEmote.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using MaplePacketLib2.Tools;
+using MapleServer2.Packets;
+using MapleServer2.Servers.Game;
+using MapleServer2.Tools;
+
+namespace MapleServer2.Types
+{
+    public class BuddyEmote
```
Comment (angelotadeucci) [inline MapleServer2/Types/BuddyEmote.cs:11]: I'm a little confused by this class. Is it just related to buddy emotes or it's an early buddy list implementation?

File: MapleServer2/PacketHandlers/Game/BuddyEmoteHandler.cs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+        private enum BuddyEmoteMode : byte
+        {
+            InviteBuddyEmote = 0x0,
+            InviteBuddyEmoteConfirm = 0x1,
+            LearnEmote = 0x2,
+            AcceptEmote = 0x3,
+            DeclineEmote = 0x4,
+            StopEmote = 0x6,
+
```
Comment (AlanMorel) [inline MapleServer2/PacketHandlers/Game/BuddyEmoteHandler.cs:29]: blank line

File: MapleServer2/PacketHandlers/Game/BuddyEmoteHandler.cs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        private void HandleAcceptEmote(GameSession session, PacketReader packet)
+        {
+            int buddyEmoteId = packet.ReadInt();
+            long senderCharacterId = packet.ReadLong();
+            CoordF senderCoords = packet.Read<CoordF>();
+            CoordF selfCoords = packet.Read<CoordF>();
+            int rotation = packet.ReadInt();
+
+
```
Comment (AlanMorel) [inline MapleServer2/PacketHandlers/Game/BuddyEmoteHandler.cs:104]: blank line

File: MapleServer2/PacketHandlers/Game/BuddyEmoteHandler.cs [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            Player buddy = GameServer.Storage.GetPlayerById(senderCharacterId);
+            if (buddy == null)
+            {
+                return;
+            }
+
+            buddy.Session.Send(BuddyEmotePacket.SendAccept(buddyEmoteId, session.Player));
+            session.Send(BuddyEmotePacket.StartEmote(buddyEmoteId, buddy.Session.Player, session.Player, selfCoords, rotation));
+            buddy.Session.Send(BuddyEmotePacket.StartEmote(buddyEmoteId, buddy.Session.Player, session.Player, selfCoords, rotation));
+
```
Comment (AlanMorel) [inline MapleServer2/PacketHandlers/Game/BuddyEmoteHandler.cs:114]: blank line

**Review by AlanMorel — CHANGES_REQUESTED** [review]

File: MapleServer2/Types/BuddyEmote.cs (hunk 1 above)
Comment (Zintixx) [reply MapleServer2/Types/BuddyEmote.cs:11]: Sorry, this was me experimenting in the beginning. This should be deleted.

**Review by angelotadeucci — APPROVED** [review]

**Pull request merged**